pkg/services/ngalert/notifier/channels/email.go

```diff
@@ -23,6 +23,7 @@ type EmailNotifier struct {
 	Message     string
 	log         log.Logger
 	tmpl        *template.Template
+	bus         bus.Bus
```

Comment:
As an alternative, you can declare a variable in this package and overwrite it in your tests.

Reply:
Agree with @santihernandezc. Specifically, if we had a test that created a new `bus.Bus` for each case, Go's test framework would run each case in parallel and tests would leak. These specific tests do not do this, but in general any time we have global state it is prone to causing hard-to-debug failures when running tests in parallel.